Tests/SonarScanner.MSBuild.PreProcessor.Test/ProcessedArgsTests.cs

```diff
@@ -361,12 +390,86 @@ public void ProcArgs_OperatingSystem_AutoDetection(PlatformOS platformOS, string
             operatingSystemProvider.OperatingSystem().Returns(_ => platformOS);
             var sut = CreateDefaultArgs(operatingSystemProvider: operatingSystemProvider);
             sut.OperatingSystem.Should().Be(expectedOperatingSystem);
+            sut.IsValid.Should().BeTrue();
+        }
+
+        [TestMethod]
+        public void ProcArgs_UserHome_ParameterProvided()
+        {
+            var directoryWrapper = Substitute.For<IDirectoryWrapper>();
+            directoryWrapper.Exists(@"C:\Users\user\.sonar").Returns(true);
+            var sut = CreateDefaultArgs(new ListPropertiesProvider([new Property(SonarProperties.UserHome, @"C:\Users\user\.sonar")]), directoryWrapper: directoryWrapper);
+            sut.UserHome.Should().Be(@"C:\Users\user\.sonar");
+            sut.IsValid.Should().BeTrue();
+            logger.AssertNoErrorsLogged();
+            logger.AssertNoWarningsLogged();
+        }
+
+        [TestMethod]
+        public void ProcArgs_UserHome_ParameterProvided_DoesNotExists()
+        {
+            var directoryWrapper = Substitute.For<IDirectoryWrapper>();
+            directoryWrapper.Exists(@"C:\Users\user\.sonar").Returns(false);
+            var sut = CreateDefaultArgs(new ListPropertiesProvider([new Property(SonarProperties.UserHome, @"C:\Users\user\.sonar")]), directoryWrapper: directoryWrapper);
+            sut.UserHome.Should().BeNull();
+            sut.IsValid.Should().BeFalse();
+            logger.AssertErrorLogged(@"The provided value for 'sonar.userHome' 'C:\Users\user\.sonar' does not exists. Specify a valid directory for 'sonar.userHome'.");
+            logger.AssertNoWarningsLogged();
+        }
+
+        [TestMethod]
+        public void ProcArgs_UserHome_Default()
+        {
+            var directoryWrapper = Substitute.For<IDirectoryWrapper>();
+            var operatingSystemProvider = Substitute.For<IOperatingSystemProvider>();
+            operatingSystemProvider.GetFolderPath(Environment.SpecialFolder.UserProfile, Environment.SpecialFolderOption.None).Returns(@"C:\Users\user");
+            directoryWrapper.Exists(@"C:\Users\user\.sonar").Returns(true);
```

Comment:
Would be nice to configure the wrapper after creation. Otherwise, when reading the code we have to change the context:
- directory setup
- os setup
- directory setup